### PR TITLE
 MS-11: Clients -List Clients

### PR DESCRIPTION
**Pull request by amacazzaga**
Task : Created List of Clients
Commit history:  inicial commit

@amacazzaga
amacazzaga committed yesterday
 
Commits on Nov 6, 2021
MS-11 list client working properly

### CODE REVIEW

File: controllers/clients.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+const getByName = (req, res) => {
+  const client = clients.find(client => client.name === req.params.name);
+  if (client) {
+    res.json(client);
+  } else {
+    res.send('Client not found');
+  }
+}
+  const getByphoneNumber = (req, res) => {
```
Comment (nicolasblasco) [inline controllers/clients.js:25]: Spaces between each function

File: controllers/clients.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+module.exports = {
+  getAll: getAll,
+  getById: getById,
+  getByName: getByName,
+  getByphoneNumber:getByphoneNumber,
+  getByCuit:getByCuit,
+  getByAddress:getByAddress,
+  getByActivity:getByActivity
+
```
Comment (nicolasblasco) [inline controllers/clients.js:88]: Fix space indentation

File: data/clients.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
+[
+  {
+    "id": 1,
```
Comment (nicolasblasco) [inline data/clients.json:3]: Change id type to string

File: controllers/clients.js (hunk 2 above)
Comment (amacazzaga) [reply controllers/clients.js]: fix!

File: controllers/clients.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+const clients = JSON.parse(fs.readFileSync("./data/clients.json"));
+
+//CLIENT LIST///
+
+const getAll = (req, res) => {
+  res.json(clients);
+};
+
+const getById = (req, res) => {
+  const client = clients.find((client) => client.id == req.params.id);
```
Comment (nicolasblasco) [inline controllers/clients.js:11]: Use ===

File: controllers/clients.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    })
+  } else {
+    res.status(404).send("Admin not found")
+  }}*/
+
+module.exports = {
+  getAll: getAll,
+  getById: getById,
+  getByName: getByName,
+  getByphoneNumber: getByphoneNumber,
```
Comment (BaltazarCortina) [inline controllers/clients.js:97]: The p en phone should be uppercase: getByPhoneNumber

File: controllers/clients.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+const getByName = (req, res) => {
+  const client = clients.find((client) => client.name === req.params.name);
+  if (client) {
+    res.json(client);
+  } else {
+    res.send("Client not found");
+  }
+};
+
+const getByphoneNumber = (req, res) => {
```
Comment (BaltazarCortina) [inline controllers/clients.js:28]: The p en phone should be uppercase: getByPhoneNumber

File: index.js [hunk 7]
```diff
@@ -1,11 +1,23 @@
...
 
 app.get('/', (req, res) => {
   res.send('Welcome to MindSet!')
 });
 
+//CLIENTS//LIST
+app.get('/clients', clientsController.getAll);
+app.get('/clients/byId/:id', clientsController.getById);
+app.get('/clients/byName/:name', clientsController.getByName);
+app.get('/clients/byphoneNumber/:phoneNumber', clientsController.getByphoneNumber);
```
Comment (BaltazarCortina) [inline index.js:14]: Same in the route: byPhoneNumber

**Review by BaltazarCortina — CHANGES_REQUESTED** [review]

**Review by damianalvarez — CHANGES_REQUESTED** [review]
@amacazzaga the PR looks good, let's solve the comments that @BaltazarCortina left and merge this one

**Review by damianalvarez — APPROVED** [review]

**Pull request merged**